### PR TITLE
Add Serial Port support in VM resource and data source

### LINKED ISSUE #25: Ability to add a serial port to a VM
Moved from https://github.com/nutanix/terraform-provider-nutanix/issues/159

The situation as reported by leonteq: 

> when using Ubuntu 16.04 cloud image, like the one you’d deploy in aws/azure/etc:
> - The image is built with enabled output on serial port; if the VM has no serial port, there’s a very long pause before it times out and moves on
> - We fixed it by removing the serial port dependency in the image
> - The other workaround is to manually create serial port on the VM and restart

Answer from Jon 

> We’ve got the ability to do this in the code already in the v3 API that will ship with AOS 5.9, which is due out later this year. When that hits, serial_port_list will be the array you’d call configure the serial_port, just like you’d do to configure a NIC with nic_list or DISK with disk_list
>  
> The particular feature that was tied too is unlikely to be backported to 5.6/5.8 (which BTW 5.8 is coming out this week), so we’ll need to wait for that API to have terraform hook into it.

This is similar to the issue about resizing disks, the API support is now there in 5.9+, and 5.10 as such, so we'll need to validate and make sure its documented correctly here.

### PR DESCRIPTION
**Pull request by marinsalinas**
closes #25

### CODE REVIEW

File: nutanix/resource_nutanix_virtual_machine.go [hunk 1]
```diff
@@ -1652,6 +1704,20 @@ func preFillResUpdateRequest(res *v3.VMResources, response *v3.VMIntentResponse)
...
+			spl[k] = &v3.VMSerialPort{
+				Index:       v.Index,
+				IsConnected: v.IsConnected,
+			}
+
+		}
+	} else {
+		spl = nil
+	}
+	res.SerialPortList = spl
```
Comment (thetonymaster) [inline nutanix/resource_nutanix_virtual_machine.go:1718]: you can do this in one step

File: nutanix/resource_nutanix_virtual_machine.go [hunk 2]
```diff
@@ -9,7 +9,7 @@ import (
 	"strings"
 	"time"
 
-	"github.com/terraform-providers/terraform-provider-nutanix/client/v3"
+	v3 "github.com/terraform-providers/terraform-provider-nutanix/client/v3"
```
Comment (thetonymaster) [inline nutanix/resource_nutanix_virtual_machine.go:12]: Isn't this import by default `v3` without the name? Why using a named import?

**Review by thetonymaster — CHANGES_REQUESTED** [review]

File: nutanix/resource_nutanix_virtual_machine.go (hunk 2 above)
Comment (marinsalinas) [reply nutanix/resource_nutanix_virtual_machine.go:12]: Is the way that go 1.12 version do the goimports

File: nutanix/resource_nutanix_virtual_machine.go (hunk 1 above)
Comment (marinsalinas) [reply nutanix/resource_nutanix_virtual_machine.go:1718]: Done @thetonymaster

**Review by thetonymaster — APPROVED** [review]

**Pull request merged**